src/main.rs

```diff
@@ -126,3 +131,40 @@ fn run_agent() -> failure::Fallible<()> {
 
     Ok(())
 }
+
+/// Deadend subcommand entry point.
+fn write_deadend_release_info(reason: Option<String>) -> failure::Fallible<()> {
+    if reason.is_some() {
+        // Avoid showing partially-written messages using tempfile and
+        // persist (rename).
+        let mut f = tempfile::Builder::new()
+            .prefix(".deadend.")
+            .suffix(".motd.partial")
+            // Create the tempfile in the same directory as the final MOTD,
+            // to ensure proper SELinux labels are applied to the tempfile
+            // before renaming.
+            .tempfile_in("/run/motd.d")
+            .with_context(|e| format!("failed to create temporary MOTD file: {}", e))?;
+        // Set correct permissions of the temporary file, before moving to
+        // the destination (`tempfile` creates files with mode 0600).
+        fs::set_permissions(f.path(), Permissions::from_mode(0o664))
```

Comment:
```suggestion
        fs::set_permissions(f.path(), Permissions::from_mode(0o644))
```

Any particular not to use `644` here?

Reply:
hmm... I just carried this change from the (previous)[https://github.com/coreos/zincati/pull/276] PR, so I don't have any strict opinion here.

Reply:
@sohankunkerkar is the resulting file owned by `root:root`? If so, this is likely not going to matter but `0644` should be enough.

Reply:
Yeah, it's owned by `root:root`
```
[core@localhost ~]$ ls -ltr /run/motd.d/85-zincati-deadend.motd 
-rw-rw-r--. 1 root root 109 Oct  9 15:43 /run/motd.d/85-zincati-deadend.motd

```